### PR TITLE
Augment session data

### PR DESCRIPTION
**Pull request by surma**
Augments the database adapters to store the expiry date and the user ID of the session token.

Note: I am actively omitting the users name and email address etc to avoid storing PII without the user potential realizing that.

### CODE REVIEW

File: src/auth/session/session-utils.ts [hunk 1]
```diff
@@ -9,23 +9,49 @@ export function sessionFromEntries(
...
+            if (typeof value === 'string') {
+              return [key, value.toString().toLowerCase() === 'true'];
+            } else if (typeof value === 'number') {
+              return [key, Boolean(value)];
+            }
+            return [key, value];
+          case 'scope':
+            return [key, value.toString()];
+          case 'expires':
+            return [key, new Date(Number(value) * 1000)];
```
Comment (paulomarg) [inline src/auth/session/session-utils.ts:38]: This is potentially null, do we need to handle that here?

File: src/auth/session/session-utils.ts [hunk 2]
```diff
@@ -36,9 +62,36 @@ const includedKeys = [
...
 ): [string, string | number][] {
-  return Object.entries(session).filter(([key]) => includedKeys.includes(key));
+  return (
+    Object.entries(session)
+      .filter(([key]) => includedKeys.includes(key))
+      // Prepare values for db storage
+      .map(([key, value]) => {
+        switch (key) {
+          case 'expires':
+            return [key, Math.floor(value.getTime() / 1000)];
```
Comment (paulomarg) [inline src/auth/session/session-utils.ts:78]: Same thing here about null expiration times

File: src/auth/session/storage/__tests__/battery-of-tests.ts [hunk 3]
```diff
@@ -20,13 +25,77 @@ export function batteryOfTests(storageFactory: () => Promise<SessionStorage>) {
...
+    );
+
+    await expect(storage.deleteSession(sessionId)).resolves.toBe(true);
+    await expect(storage.loadSession(sessionId)).resolves.toBeUndefined();
+
+    // Deleting a non-existing session should work
+    await expect(storage.deleteSession(sessionId)).resolves.toBe(true);
+  });
+
+  it('can store and delete offline sessions with user info', async () => {
```
Comment (paulomarg) [inline src/auth/session/storage/__tests__/battery-of-tests.ts:53]: Technically, offline sessions won't ever have user info, but I don't think we need any special code to block saving the sessions in that case.

File: src/auth/session/session-utils.ts (hunk 1 above)
Comment (surma) [reply src/auth/session/session-utils.ts:38]: Good catch! Added test and fix

File: src/auth/session/session-utils.ts [hunk 4]
```diff
@@ -9,23 +9,49 @@ export function sessionFromEntries(
...
+            if (typeof value === 'string') {
+              return [key, value.toString().toLowerCase() === 'true'];
+            } else if (typeof value === 'number') {
+              return [key, Boolean(value)];
+            }
+            return [key, value];
+          case 'scope':
+            return [key, value.toString()];
+          case 'expires':
+            return [key, value ? new Date(Number(value) * 1000) : undefined];
```
[the remark this answers is not in the document]
Comment (paulomarg) [reply src/auth/session/session-utils.ts:38]: Makes sense to me!

**Pull request merged**